core/src/main/java/ai/timefold/solver/core/impl/phase/custom/DefaultCustomPhase.java

```diff
@@ -74,6 +78,10 @@ public void stepEnded(CustomStepScope<Solution_> stepScope) {
 
     public void phaseEnded(CustomPhaseScope<Solution_> phaseScope) {
         super.phaseEnded(phaseScope);
+        // Only update the best solution if the custom phase first initializes the solution and made any change.
+        if (triggersFirstInitializedSolutionEvent() && !phaseScope.getStartingScore().equals(phaseScope.getBestScore())) {
```

Comment:
If the score is improved, shouldn't it _always_ give the better solution?
What would be a situation where the best solution should be ignored?

What happens if the custom phase is in between two local search phases? Is this behavior still correct?